students/MikeShand/lesson06/mailroom4_test.py

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env python3
+
+import mailroom4 as mailroom
+import sys
+import string
+import os.path
+from collections import defaultdict
+
+
+def test_donor_list():
+    mail_list = mailroom.donor_list()
+    assert "Andy" in mail_list
+    assert "Bill" in mail_list
+    assert "Chuck" in mail_list
+
+
+def test_donor_report():
+    report = mailroom.donor_report()
+    assert 'Andy' in report
+    assert 'Bill' in report
+    assert 'Chuck' in report

```

Comment:
to be a  more useful test, I would assert full report string which would test that report renders as expected vs just verifying that those names were included.